lib/symboldatabase.cpp

```diff
@@ -5470,7 +5470,7 @@ void SymbolDatabase::setValueTypeInTokenList()
             setValueType(tok, vt);
         } else if (tok->str() == "return" && tok->scope()) {
             const Function *function = tok->scope()->function;
-            if (function) {
+            if (function && function->retDef) {
```

Comment:
Is there another possibility to detect a ctor?

Reply:
I dont know., but if there are other cases besides constructors that are missing the `retDef` then this fix would still prevent the crash.

Reply:
Well, that's true.
However it should be an internal error otherwise:  `if (!function->isCtor() && !function->retDef`)?

Reply:
I imagine this could happen for overloaded cast operators also. And in old style C functions where the "int" return type is implicit.

Example C code:

    #include <stdio.h>
    main() {
      printf("hello world!\n");
      return 0;
    }